syntax = "proto3";

package TW.Theta.Proto;
option java_package = "wallet.core.jni.proto";

/// Input data necessary to create a signed transaction
message SigningInput {
    /// Chain ID string, mainnet, testnet and privatenet
    string chain_id = 1;

    /// Recipient address
    string to_address = 2;

    /// Theta token amount to send in wei (256-bit number)
    bytes theta_amount = 3;

    /// TFuel token amount to send in wei (256-bit number)
    bytes tfuel_amount = 4;

    /// Sequence number of the transaction for the sender address
    uint64 sequence = 5;

    /// Fee amount in TFuel wei for the transaction (256-bit number)
    bytes fee = 6;

    /// Private key
    bytes private_key = 7;
}

/// Transaction signing output
message SigningOutput {
    /// Signed and encoded transaction bytes
    bytes encoded = 1;

    /// Signature
    bytes signature = 2;
}
